skopt/space/space.py

```diff
@@ -69,10 +69,10 @@ def check_dimension(dimension, transform=None):
     if len(dimension) == 2:
         if any([isinstance(d, str) for d in dimension]):
             return Categorical(dimension, transform=transform)
-        elif any([isinstance(dim, float) for dim in dimension]):
```

Comment:
Since it's a little bit lengthy to modify the `check_dimension` check out the gist here:

https://gist.github.com/yngtodd/c8123a49d55a4ab0c505b3d99a472706